lib/plugins/aws/package/compile/events/apiGateway/lib/stage.js

```diff
@@ -114,6 +117,14 @@ function getLogGroupResource(service, stage) {
 }
 
 function getIamRoleResource(service, stage) {
```

Comment:
We should rename `service` here so that it's clear that this method can handle bot:

1. Referencing the correct role which is auto-created
1. Re-using the role which could be passed-in

Reply:
I committed a change, which hopefully addresses your comments.